### PR TITLE
Transmutation reagents

### PR DESCRIPTION
**Pull request by mloccy**
[ADMIN][CHEMISTRY][INTERNAL]

## About the PR 
This PR adds a new file Reagents-Transmutation.dm which groups together some of the regents that call setMaterial on turfs and objects. It also adds a new material in the form of Carpet which is just a material version of grimy carpet from floors.dmi. It also adds a new parent reagent type transmutation which has an implementation for reaction_turf and reaction_obj and a new field material_name which can be set by child types. In this PR liquid jeans and reagent carpet is brought under this new parent type becoming 

`/datum/reagent/transmutation/jeans` 
and 
`/datum/reagent/transmutation/carpet`

respectively.

It also adds a new reagent type `/datum/reagent/transmutation/custom` which essentially allows the creation at runtime of a regent that transmutes turfs and objs into any concrete material. Finally, a new admin only verb was added, Spawn-Custom-Transmutation-Reagent` that allows creation of a beaker / chem grenade filled with this new custom transmutation reagent with whatever material is desired.

## Why's this needed? 
Enables more admin shenanigans / admin ran scenarios that could be fun or interesting, and it extends the functionality of liquid carpet to objects too.

### CODE REVIEW

**Comment by mloccy** [comment]
Video showing new verb:

https://github.com/goonstation/goonstation/assets/757551/4affa12a-cdd1-4c4a-818a-d8b8c1ac86dc

File: code/modules/chemistry/Reagents-Transmutation.dm [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#define MIN_REAGENT_FOR_CONVERSION 5
+datum
```
Comment (amylizzle) [inline code/modules/chemistry/Reagents-Transmutation.dm:2]: I realise this is copy pasted from reagent code, but indent pathing is gross and hard to search. Maybe switch to absolute pathing (`/datum/reagent/transmutation`)

**Review by amylizzle — CHANGES_REQUESTED** [review]
Minor changes, otherwise lgtm

File: code/modules/admin/randomverbs.dm [hunk 2]
```diff
@@ -3034,6 +3034,46 @@ var/global/force_radio_maptext = FALSE
...
 
+/// Spawn custom reagent that can transform turfs and objs to whatever material you desire.
+/client/proc/spawn_custom_transmutation()
+	SET_ADMIN_CAT(ADMIN_CAT_FUN)
+	set name = "Spawn Custom Transmutation Reagent"
+	ADMIN_ONLY
+	var/containers = list()
+	containers["Small Beaker"] = /obj/item/reagent_containers/glass/beaker
+	containers["Large Beaker"] = /obj/item/reagent_containers/glass/beaker/large
+	containers["Grenade"] = /obj/item/chem_grenade/custom
```
[the remark this answers is not in the document]
Comment (mloccy) [reply code/modules/admin/randomverbs.dm]: Thanks

File: code/modules/chemistry/Reagents-Transmutation.dm (hunk 1 above)
Comment (mloccy) [reply code/modules/chemistry/Reagents-Transmutation.dm]: Done, I'm still getting used to dm and have been following patterns I see, but as with any old codebase I guess there's good and bad patterns. I will do this in the future, the searching problem this causes makes 100% to me, thanks for this feedback.

**Comment by ZeWaka** [comment]
x

**Pull request merged**